python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -293,6 +298,15 @@ def _scheduling_loop_step(self, topology: Topology) -> bool:
             [op.metrics for op in self._topology], {"dataset": self._dataset_tag}
         )
 
+        # Log metrics of newly completed operators.
+        for op in topology:
+            if op.completed() and not self._op_completed[op]:
+                log_str = f"Operator {op} completed. Operator Metrics:\n" + str(
+                    op._metrics.as_dict()
+                )
```

Comment:
Don't f-strings call `str` implicitly?
```suggestion
                log_str = f"Operator {op} completed. Operator Metrics:\n{op._metrics.as_dict()}"
```